lib/icinga/notification.cpp

```diff
@@ -104,8 +104,19 @@ void Notification::OnConfigLoaded()
 {
 	ObjectImpl<Notification>::OnConfigLoaded();
 
-	SetTypeFilter(FilterArrayToInt(GetTypes(), GetTypeFilterMap(), ~0));
-	SetStateFilter(FilterArrayToInt(GetStates(), GetStateFilterMap(), ~0));
+	if (!GetTypes()) {
+		// If this was created [at runtime] without specifying the "types" attribute, the SetTypes() method
+		// won't be called, consequently the type filter bitset will also be 0. Thus, we need to ensure that
+		// the type filter is set to the default value, which is all notification types.
+		SetTypes(nullptr, false, Value());
+	}
+
+	if (!GetStates()) {
+		// If this was created [at runtime] without specifying the "states" attribute, the SetStates() method
+		// won't be called, consequently the state filter bitset will also be 0. Thus, we need to ensure that
+		// the state filter is set to the default value, which is all states.
+		SetStates(nullptr, false, Value());
+	}
```

Comment:
What are the square brackets around "at runtime" supposed to say here? Does that make any difference?

Also, couldn't you this as part of the constructor? Yes, if it was actually set, it would be called twice, but with `nullptr` as array, the call is super cheap and as a bonus, the fields are always consistent, not only after this method was called (which can then be removed entirely as a bonus).